src/rpdk/core/contract/resource_client.py

```diff
@@ -115,6 +116,7 @@ def __init__(
         self._update_schema(schema)
         self._inputs = inputs
         self._timeout_in_seconds = int(timeout_in_seconds)
+        self.soft_delete = bool(soft_delete == "true")
```

Comment:
what's the diff between `bool(soft_delete == "true")` and `soft_delete == "true"`?

Reply:
I don think there is a difference. Will remove "bool"